Assets/Scripts/AdController.cs

```diff
@@ -1,21 +1,36 @@
-﻿using UnityEngine;
+using System.Collections.Generic;	// Dictionary
+using UnityEngine;
 using UnityEngine.UI;
 using UnityEngine.SceneManagement;
 
 public class AdController : ParentController {
 
 	public Button btn;
 
+	// Register server handlers
+	void Awake(){
+		_functionMapping[RequestType.LGO] = HandleLgoResponse;
+	}
+
 	// Use this for initialization
 	void Start () {
 		btn = gameObject.GetComponent<Button>();
 		btn.onClick.AddListener(TaskOnClick);
 	}
 
+	void Update(){
+		ProcessResponses();
+	}
+
 	void TaskOnClick(){
 		//Application.OpenURL("https://pbs.twimg.com/profile_images/2932495059/f664298aa5c7554496fc503909b94466_400x400.jpeg");
-		if(Server.Logout()) {
-			SceneManager.LoadSceneAsync("Startup", LoadSceneMode.Single);
-		}
+		Server.Logout(this);
```

Comment:
Should we add showing a loading circle here?  Granted logout is likely very quick but if the server communication takes longer than normal, the user could technically start going down starting a match while logout is getting processed which would be kinda weird unless I am missing something.